internal/metastore/kv/datacoord/kv_catalog.go

```diff
@@ -111,6 +111,12 @@ func (kc *Catalog) listSegments() ([]*datapb.SegmentInfo, error) {
 			return nil
 		}
 
+		// due to StatsTaskPrefix has the same prefix with SegmentPrefix, so skip it.
+		// when the WalkWithPrefix is refactored, this patch can be removed.
+		if strings.Contains(string(value), StatsTaskPrefix) {
```

Comment:
related issue https://github.com/milvus-io/milvus/issues/36178
should remove this line if issue 36178 is fixed.

Reply:
yeah， we should remove this defensive code after #36178 was fixed.